src/test/java/org/minions/devfund/noemiguzman/battleship/ShipTest.java

```diff
@@ -0,0 +1,207 @@
+     * method to test sunk.
+     */
+    @Test
+    public void testIsSunk2() {
+        final int place7 = 7;
+        Ocean ocean = new Ocean();
+        BattleShip ship = new BattleShip();
+        ship.placeShipAt(0, 0, true, ocean);
+        for (int j = 0; j < place7; j++) {
+            ocean.shootAt(0, j);
+        }
+
+        assertFalse(ship.isSunk());
+
+    }
+
+    /**
+     * method to test to string.
+     */
+    @Test
+    public void testToString() {
+        final int place8 = 8;
+        Ocean ocean = new Ocean();
+        BattleShip ship = new BattleShip();
+        ship.placeShipAt(0, 0, true, ocean);
+        for (int j = 0; j < place8; j++) {
+            ocean.shootAt(0, j);
+        }
+
+        assertEquals("x", ship.toString());
+
+    }
+
+    /**
+     * method to test to string with battleship.
+     */
+    @Test
+    public void testToStringWithShip() {
+        Ocean ocean = new Ocean();
+        final int place7 = 7;
+        BattleShip ship = new BattleShip();
+        ship.placeShipAt(0, 0, true, ocean);
+        for (int j = 0; j < place7; j++) {
+            ocean.shootAt(0, j);
+        }
+
+        assertEquals("S", ship.toString());
+
+    }
+
+    /**
+     * method to test shootat.
+     */
+    @Test
+    public void testShootAt() {
+        Ocean ocean = new Ocean();
+        BattleShip ship = new BattleShip();
+        final int column4 = 4;
+        final int column3 = 3;
+        ship.placeShipAt(0, 0, true, ocean);
+
+        assertTrue(ship.shootAt(0, column3));
+        assertTrue(ship.shootAt(0, column4));
+        assertFalse(ship.shootAt(1, column4));
+
+    }
+
+    /**
+     * method to test shoot at with battleship.
+     */
+    @Test
+    public void testShootAtWithShip() {
+        Ocean ocean = new Ocean();
+        BattleShip ship = new BattleShip();
+        final int column8 = 8;
+        final int column4 = 4;
+        final int column3 = 3;
+        ship.placeShipAt(0, 0, true, ocean);
+        for (int j = 0; j < column8; j++) {
+            ocean.shootAt(0, j);
+        }
+
+        assertFalse(ship.shootAt(0, column3));
+        assertFalse(ship.shootAt(0, column4));
+        assertFalse(ship.shootAt(1, column4));
+
+    }
+
+    /**
+     * test get ship type EmptySea.
+     */
+    @Test
+    public void testGetShipType_EmptySea() {
+        Ship ship = new EmptySea();
+        assertEquals("empty", ship.getShipType());
+    }
+
+    /**
+     * method to test shoot at EmptySea.
+     */
+    @Test
+    public void testShootAt_EmptySea() {
+        Ship ship = new EmptySea();
+        final int column4 = 4;
+
+        assertFalse(ship.shootAt(1, column4));
+
+    }
+
+    /**
+     * method to test IsSunk EmptySea.
+     */
+    @Test
+    public void testIsSunk_EmptySea() {
+        Ship ship = new EmptySea();
+        assertFalse(ship.isSunk());
+    }
+
+    /**
+     * method to test ToStrinG EmptySea.
+     */
+    @Test
+    public void testToString_EmptySea() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)